stac_ipyleaflet/stac_discovery/stac.py

```diff
@@ -23,71 +22,75 @@ class Stac:
-
-                has_cog = True if data["item_assets"] else False
-
-                id = data["id"].strip()
-                title = data["title"].strip()
-
-                start_date = data["extent"]["temporal"]["interval"][0][0]
-                end_date = data["extent"]["temporal"]["interval"][0][1]
-
-                if start_date is not None:
-                    start_date = start_date.split("T")[0]
-                else:
-                    start_date = ""
-
-                if end_date is not None:
-                    end_date = end_date.split("T")[0]
-                else:
-                    end_date = ""
-
-                bbox = ", ".join(
-                    [str(coord) for coord in data["extent"]["spatial"]["bbox"][0]]
-                )
-
-                metadata = None
-                href = None
-                for l in data["links"]:
-                    if l["rel"] == "about":
-                        metadata = l["href"]
-                    if l["rel"] == "self":
-                        href = l["href"]
-
-                description = (
-                    data["description"]
-                    .replace("\n", " ")
-                    .replace("\r", " ")
-                    .replace("\\u", " ")
-                    .replace("                 ", " ")
-                )
-
-                license = data["license"]
-                collection_obj = OutputCollectionObj(
-                    {
-                        "id": id,
-                        "title": title,
-                        "has_cog": has_cog,
-                        "start_date": start_date,
-                        "end_date": end_date,
-                        "bbox": bbox,
-                        "metadata": metadata,
-                        "href": href,
-                        "description": description,
-                        "license": license,
-                    }
-                )
-                output_collections.append(collection_obj)
-            except Exception as err:
-                error = {"error": err, "collection": collection}
-                logging.error(error)
-                return None
+        bad_collections = []
+        try:
+            for collection in collections:
+                try:
+                    data = collection.to_dict()
+                    has_cog = True if data["item_assets"] else False
+                    id = data["id"].strip()
+                    title = data["title"].strip()
+                    start_date = data["extent"]["temporal"]["interval"][0][0]
+                    end_date = data["extent"]["temporal"]["interval"][0][1]
+
+                    if start_date is not None:
+                        start_date = start_date.split("T")[0]
+                    else:
+                        start_date = ""
+
+                    if end_date is not None:
+                        end_date = end_date.split("T")[0]
+                    else:
+                        end_date = ""
```

Comment:
Wondering if this could also be optimized, to use the `collection` item:
```
  date_intervals = collection.extent.temporal.intervals
  start_date = (
      ""
      if date_intervals[0][0] is None
      else date_intervals[0][0].strftime("%Y-%m-%d")
  )
  end_date = (
      ""
      if date_intervals[0][1] is None
      else date_intervals[0][1].strftime("%Y-%m-%d")
  )
```